runtime/jcl/common/thread.cpp

```diff
@@ -589,6 +589,44 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+static UDATA findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+	if (NULL != walkState->userData1) {
+		rc = J9_STACKWALK_STOP_ITERATING;
+	} else if (NULL != walkState->method) {
+		J9JavaVM *vm = currentThread->javaVM;
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		char *methodName = (char *)J9UTF8_DATA(J9ROMMETHOD_NAME(romMethod));
+
+		if (((J9VMJAVALANGTHREAD(vm) == methodClass)
+			|| (J9VMJAVALANGVIRTUALTHREAD(vm) == methodClass)
+			|| (J9VMJDKINCUBATORCONCURRENTSCOPEDVALUECARRIER(vm) == methodClass))
+			&& (0 == strncmp(methodName, "runWith", 7))
+		) {
+			if (NULL != walkState->arg0EA) {
+				j9object_t arg = *(j9object_t *)(walkState->arg0EA - 1);
+				walkState->userData1 = (void *)arg;
+				rc = J9_STACKWALK_STOP_ITERATING;
+			} else {
+				walkState->flags |= J9_STACKWALK_ITERATE_O_SLOTS;
```

Comment:
There should be a case to disable this flag up on seeing the next method or will this guarantee that result can be found in the oslot?

Reply:
Ideally, once the appropriate frame/method is found, the walk should be stopped.

Reply:
If you tag the "found" state in one of the walkState userData fields when the O-slot iterator detects the correct object, the next invocation of the frame iterator can detect this and abort the walk.